test/FunctionalTests/Microsoft.OData.Edm.Tests/Validation/ValidationRulesTests.cs

```diff
@@ -795,6 +795,135 @@ public void SameFunctionsShouldInWithOneInReferencedModelShouldRaiseError()
+                ValidationRules.RecursiveComplexTypedPropertyMustBeOptional,
+                nestedBaseProp,
+                EdmErrorCode.RecursiveComplexTypedPropertyMustBeOptional,
+                Strings.EdmModel_Validator_Semantic_RecursiveComplexTypedPropertyMustBeOptional("nested"));
+        }
+
+        [Fact]
+        public void ComplexTypedPropertyWithSameSecondInheritedBaseTypeAsDeclaringTypeShouldError()
+        {
+            EdmComplexType baseType = new EdmComplexType("ns", "myType");
+            EdmComplexType derivedType1 = new EdmComplexType("ns", "derivedType1", baseType);
+            EdmComplexType derivedType2 = new EdmComplexType("ns", "derivedType2", derivedType1);
+            EdmComplexType derivedType3 = new EdmComplexType("ns", "derivedType3", derivedType2);
+            IEdmTypeReference derivedType3Ref = derivedType3.GetTypeReference(false);
+            IEdmStructuralProperty nestedBaseProp = baseType.AddStructuralProperty("nested", derivedType3Ref);
+
+            ValidateError(
+                ValidationRules.RecursiveComplexTypedPropertyMustBeOptional,
+                nestedBaseProp,
+                EdmErrorCode.RecursiveComplexTypedPropertyMustBeOptional,
+                Strings.EdmModel_Validator_Semantic_RecursiveComplexTypedPropertyMustBeOptional("nested"));
+        }
+
+        [Fact]
+        public void ComplexTypedPropertyWithMutualBaseTypeShouldNotError()
+        {
+            EdmComplexType baseType = new EdmComplexType("ns", "myType");
+            EdmComplexType derivedType = new EdmComplexType("ns", "derivedType", baseType);
+            EdmComplexType otherDerivedType = new EdmComplexType("ns", "otherDerivedType", baseType);
+            IEdmTypeReference derivedTypeRef = derivedType.GetTypeReference(false);
+            IEdmStructuralProperty nestedOtherDerivedProp = otherDerivedType.AddStructuralProperty("nested", derivedTypeRef);
+
+            var model = new EdmModel();
+            model.AddElement(baseType);
+            model.AddElement(derivedType);
+            model.AddElement(otherDerivedType);
+
+            ValidateNoError(
+                ValidationRules.RecursiveComplexTypedPropertyMustBeOptional,
+                model,
+                nestedOtherDerivedProp);
+        }
+
+        [Fact]
+        public void MultipleComplexTypedPropertiesWithSameTypeAsDeclaringTypeShouldAllError()
+        {
+            EdmComplexType complexType = new EdmComplexType("ns", "myType");
+            IEdmTypeReference complexTypeRef = complexType.GetTypeReference(false);
+            complexType.AddStructuralProperty("nested1", complexTypeRef);
+            complexType.AddStructuralProperty("nested2", complexTypeRef);
+
+            var model = new EdmModel();
+            model.AddElement(complexType);
+            IEnumerable<EdmError> errors;
+            List<ValidationRule> rules = new List<ValidationRule>{ValidationRules.RecursiveComplexTypedPropertyMustBeOptional};

```

Comment:
```suggestion
            List<ValidationRule> rules = new List<ValidationRule>
            {
                ValidationRules.RecursiveComplexTypedPropertyMustBeOptional
            };
```
nit: this syntax is usually written with some newlines, if you're interested in that formatting